src/main/kotlin/cacophony/pipeline/CacophonyPipeline.kt

```diff
@@ -235,4 +237,40 @@ class CacophonyPipeline(
         asm.forEach { (function, asm) -> println("$function generates asm:\n$asm") }
         return asm
     }
+
+    fun generateAsm(input: Input, dest: Path) {
+        val defs = generateAsm(input)
+        Files.newBufferedWriter(dest, StandardOpenOption.CREATE).use { writer ->
+            defs.values.forEach {
+                writer.write(it)
+                writer.newLine()
+            }
+        }
+    }
+
+    fun compile(src: Path, dest: Path) {
+        val nasm = ProcessBuilder("nasm", "-f", "elf64", "-o", dest.toString(), src.toString()).inheritIO().start()
+        nasm.waitFor().takeIf { it != 0 }?.let { status ->
+            logger?.logFailedAssembling(status)
+            throw CompileException("Unable to assemble generated code")
```

Comment:
At this stage, nasm/gcc error nearly certainly indicates a bug inside compiler, i.e. this should not be a CompileException.